ts/features/bonus/cdc/screens/CdcBonusRequestSelectResidence.tsx

```diff
@@ -20,26 +20,36 @@ import {
   cancelButtonProps,
   confirmButtonProps
 } from "../../bonusVacanze/components/buttons/ButtonConfigurations";
+import { cdcSelectedBonusSelector } from "../store/reducers/cdcBonusRequest";
+import { useIOSelector } from "../../../../store/hooks";
+import { H3 } from "../../../../components/core/typography/H3";
+import BonusIcon from "../../../../../img/features/cdc/bonus.svg";
 
-const getCheckResidencyItems = (): ReadonlyArray<RadioItem<boolean>> => [
+const getCheckResidencyItems = (): ReadonlyArray<RadioItem<residentChoice>> => [
   {
     body: I18n.t("bonus.cdc.bonusRequest.selectResidence.items.residesInItaly"),
-    id: true
+    id: "residentInItaly"
   },
   {
     body: I18n.t("bonus.cdc.bonusRequest.selectResidence.items.residesAbroad"),
-    id: false
+    id: "residentAbroad"
   }
 ];
 
+type residentChoice = "residentAbroad" | "residentInItaly";
 const CdcBonusRequestSelectResidence = () => {
   const navigation =
     useNavigation<
       IOStackNavigationProp<CdcBonusRequestParamsList, "CDC_SELECT_RESIDENCE">
     >();
   const [isResidentInItaly, setIsResidentInItaly] = React.useState<
-    boolean | undefined
-  >();
+    Record<string, residentChoice>
+  >({});
+  const cdcSelectedBonus = useIOSelector(cdcSelectedBonusSelector);
+
+  if (cdcSelectedBonus === undefined || cdcSelectedBonus.length === 0) {
```

Comment:
```suggestion
  if (cdcSelectedBonus?.length) {
```
![Schermata 2022-05-10 alle 10 37 18](https://user-images.githubusercontent.com/822471/167586538-23d3fb8d-649a-40d5-ac35-e29dc76170e1.png)
This will be true when cdcSelectedBonus is `undefined` or `length===0`

Reply:
Thank you, updated in [2c95dea](https://github.com/pagopa/io-app/pull/3949/commits/2c95deadcac4f0326f1895bb67a2e776e9f15369)